### ISSUE TITLE
27.1.3 Which Decklink driver version supported

### ISSUE THREAD

**Issue by TheSashmo** [snapcrafters/obs-studio#150]
It's not entirely clear to me which decklink driver version I need to use for obs to recognize it.

If I use apt to install obs, it works by default with whichever decklink driver that I have installed 11.2 and up.

If I use snap to install obs, it shows error that decklink iterators could not be created, drivers may not be installed, but they are.

So I suspect its a version issue?  Any suggestions?

Link to paste bin for logs: [https://pastebin.com/riT0K1Wu](url)

**Comment by ds88888888** [comment]
I wonder if you're having the same issue as me. I had Blackmagic Desktop Video version 12.1 and it was not recognising my Decklink Quad card, so I updated to version 12.2. What version do you have and has this setup worked for you before?

I am running Ubuntu 20.04.3 and my kernel version is;
<pre>$ uname -a
Linux streambox 5.11.0-40-generic #44~20.04.2-Ubuntu SMP Tue Oct 26 18:07:44 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
</pre>

I also checked that the drivers were recognised and loaded correctly as below;
<pre>$ lspci | grep Blackmagic
06:00.0 Multimedia video controller: <font color="#CC0000"><b>Blackmagic</b></font> Design DeckLink SDI/Duo/Quad
07:00.0 Multimedia video controller: <font color="#CC0000"><b>Blackmagic</b></font> Design DeckLink SDI/Duo/Quad
08:00.0 Multimedia video controller: <font color="#CC0000"><b>Blackmagic</b></font> Design DeckLink SDI/Duo/Quad
09:00.0 Multimedia video controller: <font color="#CC0000"><b>Blackmagic</b></font> Design DeckLink SDI/Duo/Quad
</pre>

<pre>$ lsmod | grep blackmagic
<font color="#CC0000"><b>blackmagic</b></font>            561152  12
</pre>

I notice in your log below, but I'm not sure how to fix it yet, but according to OBS readme notes and checking driver status it's all as expected.
<pre>20:05:03.635: A DeckLink iterator could not be created.  The DeckLink drivers may not be installed
20:05:03.635: No blackmagic support</pre>

**Comment by ds88888888** [comment]
@TheSashmo I've just uninstalled 27.1.3 custom, I can't remember why it says custom, maybe I built from source.
I installed version 27.1.3 as below, and it's successfully detecting my Blackmagic DeckLink Quad card.

<pre>$ sudo add-apt-repository ppa:obsproject/obs-studio
$ sudo apt update
$ sudo apt install -f obs-studio</pre>

**Comment by TheSashmo** [comment]
I have to give that a try, ended up going back to the other OBS not snap build to get what I needed for now.

**Comment by TheSashmo** [comment]
See, the issue is that the snap install vs. the apt install are not the same.

The apt install dosnt give me the ability to add the plugins correctly, where the snap install already comes with the plugin that I need.  BUT, the snap install dosnt support the blackmagic output.

Same OS, two versions of OBS, and one has the blackmagic, but no plugins, and the other has the plugs without blackmagic working.

**Comment by TheSashmo** [comment]
So I think we might be talking about different installs.

Theres this SNAP install from here, and the APT install from the default repo.

```
$ uname -a
Linux nxt-2sm1-038 5.11.0-40-generic #44~20.04.2-Ubuntu SMP Tue Oct 26 18:07:44 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

$ lspci | grep Blackmagic
03:00.0 Multimedia video controller: Blackmagic Design DeckLink SDI Micro
```

When doing install via SNAP it shows 27.1.3 (custom)
When doing install via APT it shows 27.1.3 (linux)

The linux version works fine with decklink drivers any version.

The custom version does not see the decklink.

**Comment by TheSashmo** [comment]
OK I think I am giving up.

snap install obs-studio

Gives me OBS 27.1.3-Modifued (linux) "decklink not working"

`(process:38466): Gtk-WARNING **: 19:48:19.157: Locale not supported by C library.
	Using the fallback 'C' locale.
Gtk-Message: 19:48:19.212: Failed to load module "canberra-gtk-module"
Gtk-Message: 19:48:19.212: Failed to load module "canberra-gtk-module"
Attempted path: share/obs/obs-studio/locale/en-US.ini
Attempted path: /usr/share/obs/obs-studio/locale/en-US.ini
Attempted path: share/obs/obs-studio/themes/Dark.qss
Attempted path: /usr/share/obs/obs-studio/themes/Dark.qss
info: User enabled --multi flag and is now running multiple instances of OBS.
info: Command Line Arguments: --multi
info: CPU Name: Intel(R) Core(TM) i7-10710U CPU @ 1.10GHz
info: CPU Speed: 1600.000MHz
info: Physical Cores: 6, Logical Cores: 12
info: Physical Memory: 7826MB Total, 4916MB Free
info: Kernel Version: Linux 5.11.0-40-generic
info: Distribution: "Ubuntu Core" "20"
info: Session Type: x11
info: Window System: X11.0, Vendor: The X.Org Foundation, Version: 1.20.11
info: Portable mode: false
Attempted path: share/obs/obs-studio/themes/Dark/no_sources.svg
Attempted path: /usr/share/obs/obs-studio/themes/Dark/no_sources.svg
QMetaObject::connectSlotsByName: No matching signal for on_tbar_position_valueChanged(int)
QMetaObject::connectSlotsByName: No matching signal for on_actionShowTransitionProperties_triggered()
QMetaObject::connectSlotsByName: No matching signal for on_actionHideTransitionProperties_triggered()
info: OBS 27.1.3-modified (linux)
info: ---------------------------------
info: ---------------------------------
info: audio settings reset:
	samples per sec: 48000
	speakers:        2
info: ---------------------------------
info: Initializing OpenGL...
info: Loading up OpenGL on adapter Intel Mesa Intel(R) UHD Graphics (CML GT2)
info: OpenGL loaded successfully, version 4.6 (Core Profile) Mesa 21.0.3, shading language 4.60
info: ---------------------------------
info: video settings reset:
	base resolution:   1280x720
	output resolution: 1280x720
	downscale filter:  Bicubic
	fps:               30/1
	format:            NV12
	YUV mode:          709/Partial
info: NV12 texture support not available
info: Audio monitoring device:
	name: Default
	id: default
info: ---------------------------------
info: [StreamFX] Loading Version 0.10.1.0-gc8484f65
info: [StreamFX] Loaded Version 0.10.1.0-gc8484f65
info: [adv-ss] version: 3b465e1fdc7feb87599c6a914c2dca815117e2e7
info: [adv-ss] curl loaded successfully
info: [adv-ss] found curl library
warning: Failed to load 'en-US' text for module: 'decklink-captions.so'
warning: Failed to load 'en-US' text for module: 'decklink-ouput-ui.so'
info: [Downstream Keyer] loaded version 0.2.1
info: [Freeze Filter] loaded version 0.3.2
info: [Gradient Source] loaded version 0.2.0
info: [pango]: Loaded system fontconfig
libDeckLinkAPI.so: cannot open shared object file: No such file or directory
warning: A DeckLink iterator could not be created.  The DeckLink drivers may not be installed
info: No blackmagic support
error: [linux-vkcapture] linux-vkcapture cannot run on non-EGL platforms
warning: Failed to initialize module 'linux-vkcapture.so'
info: [Move Transition] loaded version 2.5.1
error: os_dlopen(libnvidia-fbc.so.1->libnvidia-fbc.so.1): libnvidia-fbc.so.1: cannot open shared object file: No such file or directory

error: Unable to load NvFBC library
warning: Failed to initialize module 'nvfbc.so'
warning: Failed to load 'en-US' text for module: 'obs-audio-pan-filter.so'
info: [obs-audio-pan-filter] plugin loaded successfully (version 0.1.2)
info: [obs-browser]: Version 2.16.3
info: [obs-browser]: CEF Version 87.1.14+ga29e9a3+chromium-87.0.4280.141
error: os_dlopen(libnvidia-encode.so.1->libnvidia-encode.so.1): libnvidia-encode.so.1: cannot open shared object file: No such file or directory

info: FFMPEG VAAPI supported
warning: Failed to load 'en-US' text for module: 'obs-gphoto.so'
info: obs-gstreamer build: 9ed970d
info: [obs-ndi] hello ! (version 4.9.1)
info: [obs-ndi] Trying ''
info: [obs-ndi] Trying '/usr/lib'
info: [obs-ndi] Trying '/usr/local/lib'
info: [obs-ndi] Found NDI library at '/usr/local/lib/libndi.so.4'
info: [obs-ndi] NDI runtime loaded successfully
info: [obs-ndi] NDI library initialized successfully (NDI SDK LINUX 18:32:11 Jun 30 2020 4.5.3)
warning: Failed to load 'en-US' text for module: 'obs-vnc.so'
info: [obs-vnc] plugin loaded successfully (version 0.4.0)
info: [obs-websocket] you can haz websockets (version 4.9.1)
info: [obs-websocket] qt version (compile-time): 5.15.2 ; qt version (run-time): 5.15.2
info: [obs-websocket] module loaded!
info: [Replay Source] loaded version 1.6.9
info: [Source Copy] loaded version 0.1.3
info: [Source Record] loaded version 0.2.6
info: [Source Switcher] loaded version 0.3.0
info: [spectralizer] Loading v1.3.2 build time 2021.10.29 12:19
info: [TimeWarpScan] loaded version 0.1.6
info: [Transition Table] loaded version 0.2.2
info: VLC found, VLC video source enabled
info: ---------------------------------
info:   Loaded Modules:
info:     vlc-video.so
info:     transition-table.so
info:     time-warp-scan.so
info:     text-freetype2.so
info:     spectralizer.so
info:     source-switcher.so
info:     source-record.so
info:     source-copy.so
info:     sndio.so
info:     rtmp-services.so
info:     replay-source.so
info:     recursion-effect.so
info:     obs-x264.so
info:     obs-websocket.so
info:     obs-vst.so
info:     obs-vnc.so
info:     obs-transitions.so
info:     obs-rtspserver.so
info:     obs-rgb-levels-filter.so
info:     obs-outputs.so
info:     obs-ndi.so
info:     obs-libfdk.so
info:     obs-gstreamer.so
info:     obs-gphoto.so
info:     obs-filters.so
info:     obs-ffmpeg.so
info:     obs-browser.so
info:     obs-audio-pan-filter.so
info:     nvfbc.so
info:     move-transition.so
info:     linux-vkcapture.so
info:     linux-v4l2.so
info:     linux-pulseaudio.so
info:     linux-jack.so
info:     linux-decklink.so
info:     linux-capture.so
info:     linux-alsa.so
info:     libtext-pango.so
info:     image-source.so
info:     gradient-source.so
info:     frontend-tools.so
info:     freeze-filter.so
info:     dynamic-delay.so
info:     dvd-screensaver.so
info:     downstream-keyer.so
info:     dir-watch-media.so
info:     decklink-ouput-ui.so
info:     decklink-captions.so
info:     advanced-scene-switcher.so
info:     StreamFX.so
info: ---------------------------------
info: ==== Startup complete ===============================================
info: All scene data cleared
info: ------------------------------------------------
info: pulse-input: Server name: 'pulseaudio 13.99.1'
info: pulse-input: Audio format: s16le, 44100 Hz, 2 channels
info: pulse-input: Started recording from 'alsa_output.pci-0000_00_1f.3.analog-stereo.monitor'
info: [Loaded global audio device]: 'Desktop Audio'
info: pulse-input: Server name: 'pulseaudio 13.99.1'
info: pulse-input: Audio format: s16le, 44100 Hz, 2 channels
info: pulse-input: Started recording from 'alsa_input.pci-0000_00_1f.3.analog-stereo'
info: [Loaded global audio device]: 'Mic/Aux'
info: Switched to scene 'Scene'
info: [adv-ss] trying to reconnect to  in 10 seconds.
info: ------------------------------------------------
info: Loaded scenes:
info: - scene 'Scene':
info: ------------------------------------------------
info: [adv-ss] started
info: [obs-websocket] WSServer::start: Not locked to IPv4 bindings
info: [obs-websocket] server started successfully on port 4444
info: [obs-websocket] IO thread started.
Attempted path: share/obs/obs-studio/images/overflow.png
Attempted path: /usr/share/obs/obs-studio/images/overflow.png
info: adding 42 milliseconds of audio buffering, total audio buffering is now 42 milliseconds (source: Mic/Aux)
`

**Comment by flexiondotorg** [comment]
Duplicate of #40

**Comment by TheSashmo** [comment]
I see that you closed this and its a duplicate of #40, But in 40 it doesn't confirm or inform what the actual solution is other than being patched, but its not in snap, and there are not clear instructions on where/how that needs to be resolved.  Maybe I am just not looking/reading it deep enough, but can someone clarify that?
